node/test/go/aws-lambda-sdk/integration.test.js

```diff
@@ -0,0 +1,225 @@
+'use strict';
+
+const { expect } = require('chai');
+
+const path = require('path');
+const log = require('log').get('test');
+const { TracePayload } = require('@serverless/sdk-schema/dist/trace');
+const cleanup = require('./lib/cleanup');
+const createCoreResources = require('./lib/create-core-resources');
+const basename = require('./lib/basename');
+const getProcessFunction = require('../../lib/get-process-function');
+const resolveOutcomeEnumValue = require('../../utils/resolve-outcome-enum-value');
+const resolveNanosecondsTimestamp = require('../../utils/resolve-nanoseconds-timestamp');
+const normalizeEvents = require('../../utils/normalize-events');
+const spawn = require('child-process-ext/spawn');
+const resolveTestVariantsConfig = require('../../lib/resolve-test-variants-config');
+const resolveDirZipBuffer = require('../../utils/resolve-dir-zip-buffer');
+
+const fixturesDirname = path.resolve(__dirname, '../../../../go/packages/slslambda/fixtures');
+
+for (const name of ['TEST_EXTERNAL_LAYER_FILENAME']) {
+  // In tests, current working directory is mocked,
+  // so if relative path is provided in env var it won't be resolved properly
+  // with this patch we resolve it before cwd mocking
+  if (process.env[name]) process.env[name] = path.resolve(process.env[name]);
+}
+
+describe('Go: integration', function () {
+  this.timeout(120000);
+  const coreConfig = {};
+
+  function buildZipFile(functionName, architecture) {
+    return async () => {
+      const script = path.resolve(__dirname, 'build-go-binary.sh');
+      const moduleDir = path.resolve(fixturesDirname, functionName);
+      await spawn(script, [moduleDir, architecture]);
+      const outputDir = path.resolve(moduleDir, 'build', architecture);
+      const zipFile = await resolveDirZipBuffer(outputDir);
+      const isArm64 = architecture === 'arm64';
+      return {
+        Runtime: isArm64 ? 'provided.al2' : 'go1.x',
+        Architectures: isArm64 ? ['arm64'] : ['x86_64'],
+        Code: { ZipFile: zipFile },
+      };
+    };
+  }
+
+  const useCasesConfig = new Map([
+    [
+      'success',
+      {
+        variants: new Map([
+          [
+            'go1-x',
+            {
+              deferredConfiguration: buildZipFile('success', 'amd64'),
+            },
+          ],
+          [
+            'provided-al2',
+            {
+              deferredConfiguration: buildZipFile('success', 'arm64'),
+            },
+          ],
+        ]),
+      },
+    ],
+  ]);
+
+  const testVariantsConfig = resolveTestVariantsConfig(useCasesConfig, {
+    commonTestConfig: {
+      configuration: { Handler: 'bootstrap' },
```

Comment:
So with Go SDK, user is required to build the handler together with our SDK? We cannot inject in similar non-destructive way as it's with Node.js and Python, right?

Reply:
Yes, exactly.
And `bootstrap` handler name is a requirement of `provided.al2` Lambda runtime. The rule is:
* Amazon Linux 1: `go1.x` runtime
* Amazon Linux 2: `provided.al2` runtime
> As one of the later supported runtimes for Lambda, Go is implemented differently than other native runtimes. Under the hood, Go is treated as a custom runtime and runs accordingly. A Go developer can take advantage of this by choosing the provided.al2 runtime and providing the required bootstrap file. 

https://aws.amazon.com/blogs/compute/migrating-aws-lambda-functions-to-al2/

Reply:
Great thanks for clarifications 👍